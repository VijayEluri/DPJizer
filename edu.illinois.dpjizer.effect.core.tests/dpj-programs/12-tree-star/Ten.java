/**
 * This file is licensed under the University of Illinois/NCSA Open Source License. See LICENSE.TXT for details.
 */
/**
 * C.m1 -> C.m2
 * C.m2 -> C.m1 where {X/X:L}
 * C.m2 -> C.m3 
 * 
 * C.m1: X
 * C.m3: X:L
 */

package ten;

/**
 * 
 * @author Mohsen Vakilian
 * 
 */
class C<region X> {

	int a in X;
	region L;
	int b in X:L;
	C<X:L> node in X:L;

	C() {}

	void m1() {
		if (a == 1) return;
		m2();
	}

	void m2() {
		node.m1();
		m3();
	}

	void m3() {
	}
}