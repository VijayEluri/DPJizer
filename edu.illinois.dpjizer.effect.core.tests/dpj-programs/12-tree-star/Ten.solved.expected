ten.C.C() pure
ten.C.m1() reads X, X : ten.C.L : *
ten.C.m2() reads X : ten.C.L : *
ten.C.m3() pure
